src/handlers/crowdloan.contributed.event.ts

```diff
@@ -5,10 +5,52 @@ import {
   SubstrateNetwork,
 } from '../model';
 import { encodeAddress, getRegistry } from '../utils/registry';
-import { CrowdloanContributedEvent } from '../types/polkadot/events';
+import { CrowdloanContributedEvent as PolkadotCrowdloanContributedEvent } from '../types/polkadot/events';
+import { CrowdloanContributedEvent as KusamaCrowdloanContributedEvent } from '../types/kusama/events';
 import { getOrCreate } from '../utils/store';
 import getAccountHex from '../utils/getAccountHex';
 
+interface ContributedEvent {
+  address: Uint8Array;
+  paraId: number;
+  amount: bigint;
+}
+
+function getContributedEvent(
+  ctx: EventHandlerContext,
+  network: SubstrateNetwork
+): ContributedEvent {
+  switch (network) {
+    case SubstrateNetwork.kusama: {
+      const event = new PolkadotCrowdloanContributedEvent(ctx);
+
+      if (event.isV9110) {
+        const [address, paraId, amount] = event.asV9110;
+        return { address, paraId, amount };
+      } else {
+        const [address, paraId, amount] = event.asLatest;
+        return { address, paraId, amount };
+      }
+    }
+
+    case SubstrateNetwork.polkadot: {
+      const event = new KusamaCrowdloanContributedEvent(ctx);
+
+      if (event.isV9010) {
```

Comment:
Can this duplicated logic be moved out of the switch statement to a function?

Could we reduce it down to the following ?

```
const [address, paraId, amount] = event.isV9110 ? event.asV9110 : event.asLatest;
return { address, paraId, amount };
```

Reply:
Just noticed the one character difference between them.

Reply:
We can, but new versions will be released and we'd have to refactor. I think we can make it less repetitive though:

```typescript
case SubstrateNetwork.kusama: {
      const event = new PolkadotCrowdloanContributedEvent(ctx);
      let eventData

      if (event.isV9110) {
        eventData = event.asV9110;
      } else {
        eventData = event.asLatest;
      }
      
      return {
        address: eventData[0],
        paraId: eventData[1],
        amount: eventData[2],
      }
    }
```